File: Back/User/src/main/java/com/user/management/User/auth/RegisterRequest.java
package com.user.management.User.auth;

import com.user.management.User.user.Role;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class RegisterRequest {
    private String username;
    private String password;
    private String email;
    private String className;
    private String department;
    private String managedService;
    private Role role;
    private boolean mfaEnabled;

}

File: Front/SwifthubFront/src/app/BackOffice/Model/RegisterRequest.ts
import {Role} from "../../enums/role";

export interface RegisterRequest{

  username? : string;
  password? : string;
  email? : string;
  className? : string;
  department? : string;
  managedService? : string;
  role?: Role;
  mfaEnabled?: String;
}

